### PR TITLE
MAINT: Datetime deprecations

### LINKED ISSUE #1322: datetime deprecations in Python 3.12
There appear to be a few cases:
`datetime.utcnow`:
```
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/commands/publish.py:285: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    'timestamp': util.datetime_to_js_timestamp(datetime.datetime.utcnow())
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/feed.py:124: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    updated = datetime.datetime.utcnow()
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/results.py:471: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    started_at = datetime.datetime.utcnow()
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/runner.py:92: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    started_at = datetime.datetime.utcnow()
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/runner.py:276: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    started_at = datetime.datetime.utcnow()
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/runner.py:308: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    duration = (datetime.datetime.utcnow() - started_at).total_seconds()
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/runner.py:310: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    started_at = datetime.datetime.utcnow()
  /builddir/build/BUILDROOT/asv-0.6.0-1.fc40.x86_64/usr/lib64/python3.12/site-packages/asv/runner.py:354: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    ended_at = datetime.datetime.utcnow()
  /builddir/build/BUILD/asv-0.6.0/test/test_results.py:28: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    timestamp1 = datetime.datetime.utcnow()
  /builddir/build/BUILD/asv-0.6.0/test/test_run.py:137: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    timestamp = util.datetime_to_js_timestamp(datetime.datetime.utcnow())
  /builddir/build/BUILD/asv-0.6.0/test/test_runner.py:57: DeprecationWarning: datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.now(datetime.UTC).
    start_timestamp = datetime.datetime.utcnow()
```
and `datetime.utcfromtimestamp`:
```
  /builddir/build/BUILD/asv-0.6.0/test/test_util.py:355: DeprecationWarning: datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.fromtimestamp(timestamp, datetime.UTC).
    t = datetime.datetime.utcfromtimestamp(ts)
  /builddir/build/BUILD/asv-0.6.0/test/test_util.py:375: DeprecationWarning: datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.fromtimestamp(timestamp, datetime.UTC).
    t = datetime.datetime.utcfromtimestamp(ts)
```

### PR DESCRIPTION
**Pull request by HaoZeke**
Closes #1322.

### CODE REVIEW

**Comment by mattip** [comment]
Is there a CI run with 3.12rc1?

**Comment by HaoZeke** [comment]
> Is there a CI run with 3.12rc1?

We do now :) I think this is good to go.

File: asv/benchmarks.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def iter_unique(iter):
                     try:
                         with open(result_file, 'r') as fp:
                             benchmarks = json.load(fp)
-                    except (IOError, ValueError):
+                    except (OSError, ValueError):
```
Comment (mattip) [inline asv/benchmarks.py:201]: I think these should be split into two at some point. The errors are very different, one is that the expected file cannot be found, the other is something else (what?) that fails during discovery.

**Comment by mattip** [comment]
I think it is fine to drop pypy3.8 testing. The version is not supported upstream, and the test is very slow.

Other than one small nit, not really connected to this PR, it all looks good, thanks. I will merge as-is, the two comments can be dealt with elsewhere.

**Pull request merged**